### ISSUE TITLE
update_forward_refs method is deprecated

### ISSUE THREAD

**Issue by kjwpaulson** [koxudaxi/datamodel-code-generator#1440]
**Describe the bug**
The generated model may add something like `MyModel.update_forward_refs()`

pydantic reports a warning something similar to:
```
model.py:3937
  model.py:3937: PydanticDeprecatedSince20: The `update_forward_refs` method is deprecated; use `model_rebuild` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.0.3/migration/
    MyModel.update_forward_refs()
```

Example schema:
I can't share the schema used, but I can create something if it helps.

Used commandline:
```
    generate(
        json_schema,
        output=out_file,
        input_filename=in_file.name,
        input_file_type=InputFileType.OpenAPI,
        output_model_type=DataModelType.PydanticV2BaseModel,
        snake_case_field=True,
        allow_population_by_field_name=True,
        force_optional_for_required_fields=True,  # force-optional
        disable_timestamp=True,
        use_schema_description=True,
        capitalise_enum_members=True,
        use_subclass_enum=True,
        target_python_version=PythonVersion.PY_38,
        validation=True,
        use_double_quotes=True,
        encoding="utf_8",
    )
```

**Expected behavior**
No depreciation warning by using `model_rebuild` instead.

**Version:**
```
datamodel-code-generator version: 0.21.2
                pydantic version: 2.0.3
           pydantic-core version: 2.3.0 release build profile
                  python version: 3.11.2 (main, May 30 2023, 17:45:26) [GCC 12.2.0]
                        platform: Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.37
        optional deps. installed: ['email-validator', 'typing-extensions']
```

**Comment by ofek** [comment]
We are getting this warning as well

**Comment by koxudaxi** [comment]
Thank you for sharing the problem.
I will fix it

**Comment by lmmx** [comment]
This appears to be the product of `dump_resolve_reference_action`, so we should add a `dump_resolve_reference_action` for the v2 model and supply this instead based on the output type.

The flow looks like:

- pass the method `dump_resolve_reference_action` based on the `dump_resolve_reference_action` of the model type

https://github.com/koxudaxi/datamodel-code-generator/blob/89a7640f91330a95bddb4b5512077897e3258ca3/datamodel_code_generator/__init__.py#L365-L377

This is still set identically to the pydantic (v1):

https://github.com/koxudaxi/datamodel-code-generator/blob/89a7640f91330a95bddb4b5512077897e3258ca3/datamodel_code_generator/model/__init__.py#L35-L42

i.e.

https://github.com/koxudaxi/datamodel-code-generator/blob/89a7640f91330a95bddb4b5512077897e3258ca3/datamodel_code_generator/model/pydantic/__init__.py#L13-L16

We should add an alternative to the pydantic v2 module.

```py
def dump_resolve_reference_action(class_names: Iterable[str]) -> str:
    return '\n'.join(
        f'{class_name}.model_rebuild()' for class_name in class_names
    )
```

Confirmed this eliminates the warning, sending PR now.

**Comment by koxudaxi** [comment]
The issue has been fixed in 0.21.3.
